File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> ans = new ArrayList<>();

    public List<Integer> rightSideView(TreeNode root) {
        if(root == null){
            return ans;
        }
        helper(root, 0);
        return ans;
    }

    void helper(TreeNode node, int level){
        if(level == ans.size()){
            ans.add(node.val);
        }
        
        if(node.right != null){
            helper(node.right, level + 1);
        }
        if(node.left != null){
            helper(node.left, level + 1);
        }

    }
}

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):

        self.ans = []

    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return self.ans
        self.helper(root, 0)
        return self.ans

    def helper(self, node, level):

        if level == len(self.ans):
            self.ans.append(node.val)

        if node.right:
            self.helper(node.right, level + 1)

        if node.left:
            self.helper(node.left, level + 1)

